client/jetpack-cloud/components/sidebar/header/index.tsx

```diff
@@ -0,0 +1,53 @@
+import { useTranslate } from 'i18n-calypso';
+import { useCallback } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import AllSites from 'calypso/blocks/all-sites';
+import Site from 'calypso/blocks/site';
+import { setLayoutFocus } from 'calypso/state/ui/layout-focus/actions';
+import getSelectedSiteId from 'calypso/state/ui/selectors/get-selected-site-id';
+import JetpackLogo from './jetpack-logo.svg';
+
+type Props = {
+	forceAllSitesView?: boolean;
+};
+
+const AllSitesIcon = () => (
+	<img
+		className="jetpack-cloud-sidebar__all-sites-icon"
+		src={ JetpackLogo }
+		alt=""
+		role="presentation"
+	/>
+);
+
+const Header = ( { forceAllSitesView = false }: Props ) => {
```

Comment:
What I thought for us to make the header usable for use cases other than Site selection, what do you think of just passing the icon, label, and onClick event as props and moving the site logic to the consuming component? In this case, the `jetpack-cloud/sidebar` component will be responsible for wiring everything.

Though this would require more effort to recreate the site logic for pulling site data (URL, images, etc.), this aligns with our primary goal of making a reusable component for the rest of the teams.

Reply:
I'm agreeable with this idea, but poking around, it looks like it'll take quite a bit of restructuring to accomplish (especially once we have the `Navigator` architecture in place). How do you feel about us doing this in a separate PR?

Reply:
Yes, that sounds good to me 👍